compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -245,11 +248,7 @@ internal class BindingModuleGenerator(
               bindings
             } else {
               bindings
-                .groupBy {
-                  // It's safe to use the FqName here, because we don't allow generic super types.
-                  // If we wouldn't have this limitation, we'd need to use the Kotlin type.
-                  it.boundType.fqNameSafe
```

Comment:
That was the issue, we only looked at the class name and completely forgot about qualifiers.